lib/http_clients/b2b.ex

```diff
@@ -31,35 +33,23 @@ defmodule HttpClients.B2b do
     %Partner{id: id, name: name, slug: slug, api_key: api_key}
   end
 
-  @spec client(String.t(), String.t(), String.t(), String.t() | nil) :: Tesla.Client.t()
-  def client(base_url, client_id, client_secret, api_key \\ nil) do
-    headers = authorization_headers(client_id, client_secret, api_key)
+  def client(config) do
+    config
+    |> Enum.reduce(default_config(), fn item, acc -> [add_middleware(item) | acc] end)
+    |> Tesla.client()
+  end
+
+  defp add_middleware({:url, url}), do: {Tesla.Middleware.BaseUrl, url}
```

Comment:
na linha do comentário do @willianschelles, será que não fica mais simples assim?
```suggestion
  def client(%{url: base_url, authorization: authorization} = config) do
    Tesla.client([
      {Tesla.Middleware.BaseUrl, base_url},
      Tesla.Middleware.JSON,
      {Tesla.Middleware.Headers, authorization_headers(authorization)},
      {Tesla.Middleware.Retry, delay: 1_000, max_retries: 3},
      {Tesla.Middleware.Timeout, timeout: 15_000},
      {Tesla.Middleware.Logger, filter_headers: ["Authorization"]},
      Goodies.Tesla.Middleware.RequestIdForwarder
    ])
  end
  
  def authorization_headers(%{bearer: token}), do: [{"Authorization", "Bearer #{token}"}]

  def authorization_headers(%{basic: %{user: user, password: password}}) do
    [{"Authorization", "Basic #{Base.encode64("#{user}:#{password}")}"}]
  end
```
ai não tem necessidade do modulo Authorization